src/components/tl/mlx5/mcast/tl_mlx5_mcast_one_sided_reliability.c

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5_mcast_one_sided_reliability.h"
+
+static ucc_status_t ucc_tl_mlx5_mcast_one_sided_setup_reliability_buffers(ucc_base_team_t *team)
+{
+    ucc_tl_mlx5_team_t            *tl_team  = ucc_derived_of(team, ucc_tl_mlx5_team_t);
+    ucc_status_t                   status   = UCC_OK;
+    ucc_tl_mlx5_mcast_coll_comm_t *comm     = tl_team->mcast->mcast_comm;
+    int                            one_sided_total_slots_size, i;
+
+    /* this array keeps track of the number of recv packets from each process
+     * used in all the protocols */
+    comm->recv_list = ucc_calloc(1, comm->commsize * sizeof(uint32_t), "recv_list");
+    if (!comm->recv_list) {
+        tl_error(comm->lib, "unable to malloc for recv_list");
+        status = UCC_ERR_NO_MEMORY;
+        goto failed;
+    }
+
+    /* this array holds all the remote-addr/rkey of sendbuf from processes in the team 
+     * used in sync design. Needs to be set during each mcast-allgather call after sendbuf 
+     * registeration */
+    comm->slot_mem_info_list = ucc_calloc(1, comm->commsize * sizeof(ucc_tl_mlx5_mcast_slot_mem_info_t),
+                                          "slot_mem_info_list");
+    if (!comm->slot_mem_info_list) {
+        tl_error(comm->lib, "unable to malloc for slot_mem_info_list");
+        status = UCC_ERR_NO_MEMORY;
+        goto failed;
+    }
+
+    /* below data structures are used in async design only */
```

Comment:
what is sync/async design?
If they are only used in one case, why not skeeping this code segment in the other case?